assets/js/modules/tagmanager/components/setup/SetupMain.js

```diff
@@ -72,7 +67,7 @@ export default function SetupMain( { finishSetup } ) {
 	} else if ( isCreateAccount || ! accounts?.length ) {
 		viewComponent = <AccountCreate />;
 	} else {
-		viewComponent = <SetupForm finishSetup={ finishSetupAndNavigate } />;
+		viewComponent = <SetupForm finishSetup={ finishSetup } setIsNavigating={ setIsNavigating } />;
```

Comment:
I'm not a fan of splitting this up here, it feels a bit counter-intuitive, especially when compared with the other module setup code, especially since `setIsNavigating` is a quite low-level implementation detail. I think we could do the following instead:
* Keep this component here unchanged. Whenever the setup is finished, `isNavigating` here should also be set, similar to today.
* In `SetupForm`, handle its own `isNavigating` local state and render a `ProgressBar` if applicable. This keeps the logic more separate between the two components.

Reply:
This sounds reasonable but the problem with that is that we currently set `isNavigating` to true at the beginning to prevent flashes between logic that would otherwise show a progress bar (even when successful – something that is visible today) and then only set it back to false if there's a problem. If we were to duplicate the progress bar rendering between components that seems more fragile to visual inconsistencies, when switching from one component to another (this may only happen for a split second or two but will likely happen under normal circumstances). For example, the vertical/horizontal alignment should be exactly the same between the two components (which is an odd coupling to have) but also it would look a bit odd for the position of the gradient/animation to snap to a different position due to rendering a different progress bar in the same place.

For this reason, I think it's preferable to keep the single progress bar in the parent. As far as sharing low-level functions between components, I'm not sure I agree with that entirely. `setIsNavigating` is just setting some state that a navigation is happening. This state happens to cause the progress bar to be rendered. If the shared state was `setProgressBarActive` – that would be much more low-level I think.

As an alternative, I think we can use form state in the datastore to avoid needing to pass this as a prop. That way `finishSetup` can remain the same implementation all the way through as well (decorating this is a bit confusing IMO even though I did that 😄 ). Let me try this and see what you think.